### ISSUE TITLE
Contrib/ThermostatDevice : NameError: name 'self' is not defined

### ISSUE THREAD

**Issue by Poil** [jasonacox/tinytuya#205]
Hi,

I try to use the example, it gives me this error
```
Traceback (most recent call last):
  File "./test.py", line 14, in <module>
    tstatdev = Contrib.ThermostatDevice( 'xxxx', '192.168.3.36', 'xxxxx' )
  File "/home/tuya/.local/lib/python3.8/site-packages/tinytuya/Contrib/ThermostatDevice.py", line 271, in __init__
    self.schedule = None
NameError: name 'self' is not defined
```

tinytuya==1.7.1
python 3.8.10

Best regards

**Comment by jasonacox** [comment]
Thanks for opening this @Poil !  

@uzlonewolf Is it possible that we broke this with the the new `def __init__()` approach (here https://github.com/jasonacox/tinytuya/commit/ebd2a0e91a3d2a0577ef5c3b1375c29f4ef08367)?  This is the test from the example:

```python
from tinytuya import Contrib
tstatdev = Contrib.ThermostatDevice( 'abcdefghijklmnop123456', '172.28.321.475', '1234567890123abc' )
```

**Comment by jasonacox** [comment]
Update: I had to make this change to get it to work:

```python
    def __init__(self, *args, **kwargs):
        # set the default version to 3.3 as there are no 3.1 devices
        if 'version' not in kwargs or not kwargs['version']:
            kwargs['version'] = 3.3
        # set persistant so we can receive sensor broadcasts
        if 'persist' not in kwargs:
            kwargs['persist'] = True
        super(ThermostatDevice, self).__init__(*args[1:], **kwargs)
```

or this

```python
    def __init__(*args, **kwargs):
        self = args[0]
        # set the default version to 3.3 as there are no 3.1 devices
        if 'version' not in kwargs or not kwargs['version']:
            kwargs['version'] = 3.3
        # set persistant so we can receive sensor broadcasts
        if 'persist' not in kwargs:
            kwargs['persist'] = True
        super(ThermostatDevice, args[0]).__init__(*args[1:], **kwargs)
```

**Comment by uzlonewolf** [comment]
Not sure how I missed this...  Changing to `def __init__(self, *args, **kwargs):` is cleaner, but I thought I tried it back when I first did this and ran into a problem.  Of course now I don't remember what the problem was...  Only thing I could find this time is it makes the exception confusing if you forget to include the dev_id; to fix that I ended up moving the socket close in `__del__()` into a try/except block.

**Comment by jasonacox** [comment]
Thank you @uzlonewolf !!  And thank you @Poil for finding this! I will publish the update as 1.7.2 later this weekend.

**Comment by jasonacox** [comment]
Fix has been pushed as [1.7.2](https://pypi.org/project/tinytuya/1.7.2/) - thanks again for finding this @Poil and thanks for the fix @uzlonewolf .
